tests/test_elementseq.py

```diff
@@ -82,44 +81,68 @@ def test_mul_references(self):
   def test_children(self):
     self.assertEqual(tuple(self.seq.children), tuple(itertools.chain.from_iterable(ref.child_refs for ref in self.check)))
 
+  def test_edges(self):
+    self.assertEqual(tuple(self.seq.edges), tuple(itertools.chain.from_iterable(ref.edge_refs for ref in self.check)))
+
   def test_getpoints(self):
     self.assertEqual(self.seq.getpoints('bezier', 2), tuple(ref.getpoints('bezier', 2) for ref in self.check))
 
-common(
-  'PlainReferences',
-  seq=nutils.elementseq.PlainReferences([square, triangle], 2),
-  check=[square, triangle],
-  checkndims=2)
-common(
-  'UniformReferences',
-  seq=nutils.elementseq.UniformReferences(square, 3),
-  check=[square]*3,
-  checkndims=2)
-common(
-  'SelectedReferences',
-  seq=nutils.elementseq.SelectedReferences(nutils.elementseq.PlainReferences([square, triangle, square], 2), [1, 2]),
-  check=[triangle, square],
-  checkndims=2)
-common(
-  'ChainedReferences',
-  seq=nutils.elementseq.ChainedReferences([nutils.elementseq.PlainReferences([square, triangle], 2)]*2),
-  check=[square, triangle]*2,
-  checkndims=2)
-common(
-  'RepeatedReferences',
-  seq=nutils.elementseq.RepeatedReferences(nutils.elementseq.PlainReferences([square, triangle, square], 2), 2),
-  check=[square, triangle, square]*2,
-  checkndims=2)
-common(
-  'ProductReferences',
-  seq=nutils.elementseq.ProductReferences(nutils.elementseq.PlainReferences([square, triangle], 2), nutils.elementseq.PlainReferences([line, line], 1)),
-  check=[square*line]*2+[triangle*line]*2,
-  checkndims=3)
-common(
-  'ChildReferences',
-  seq=nutils.elementseq.ChildReferences(nutils.elementseq.PlainReferences([square, triangle], 2)),
-  check=[square]*4+[triangle]*4,
-  checkndims=2)
+class EmptyReferences(TestCase, Common):
+  @classmethod
+  def setUpClass(cls):
```

Comment:
why not just setUp instead of setUpClass? (applies to all following)

Reply:
done